### PR TITLE
Dev 872 add nft uri to the close bounty action

### PR DESCRIPTION
**Pull request by clarkjoao**

### CODE REVIEW

File: pages/api/nft/index.ts [hunk 1]
```diff
@@ -1,68 +1,106 @@
...
   
+    const participants = await Promise.all(proposal.details.map(async(detail: ProposalDetail) => {
+      if(!detail.recipient) return;
+
+      const user = await models.user.findOne({
+        where: { address: {
+        [Op.iLike]: String(detail.recipient)
+        } }});
+
+      const distributedAmount = networkBounty.tokenAmount * detail.percentage / 100;
```
Comment (vhcsilva) [inline pages/api/nft/index.ts:74]: use the function [calculateDistributedAmounts](https://github.com/bepronetwork/web-network/blob/dev/helpers/calculateDistributedAmounts.ts) to get the real paid amount

**Review by vhcsilva — CHANGES_REQUESTED** [review]

File: pages/api/nft/index.ts [hunk 2]
```diff
@@ -1,68 +1,106 @@
...
+    if (!await DAOService.start()) return res.status(500).json("Failed to connect with chain");
+    if(!await DAOService.loadNetwork(customNetwork.networkAddress)) 
+      return res.status(500).json("network could not be loaded");
+
+    const network = DAOService.network;
+
+    await network.start();
+    
+    const networkBounty = await network.getBounty(issueContractId);
+    if (!networkBounty) return res.status(404).json("Invalid");
```
Comment (moshmage) [inline pages/api/nft/index.ts:59]: missing "nft already minted" logic

**Review by moshmage — CHANGES_REQUESTED** [review]

File: pages/api/nft/index.ts [hunk 3]
```diff
@@ -1,68 +1,106 @@
...
 
-  if(!hash) return res.status(500);
+    const proposal = networkBounty.proposals.find(p=> p.id === +proposalscMergeId)
 
-  const url = `${defaultConfig.urls.ipfs}/${hash}`;
+    if(!proposal)
```
Comment (moshmage) [inline pages/api/nft/index.ts:63]: missing "proposal not disputed" logic, along with "proposal can be accepted"

File: pages/api/nft/index.ts [hunk 4]
```diff
@@ -1,68 +1,106 @@
...
+      
+      return { 
+        githubHandle: user?.githubHandle || '', 
+        percentage: detail.percentage, 
+        address: detail.recipient, 
+        distributedAmount,
+      };
+    }))
+
+    const pullRequest = networkBounty.pullRequests.find(pr=> pr.id === proposal.prId)
```
Comment (moshmage) [inline pages/api/nft/index.ts:84]: missing denied/canceled/error logic for PR

File: pages/api/nft/index.ts [hunk 5]
```diff
@@ -1,68 +1,106 @@
...
+
+    if(!hash) return res.status(500);
+
+    const url = `${defaultConfig.urls.ipfs}/${hash}`;
  
-  return res.status(200).json({url});
+    return res.status(200).json({url});
+  }
+  catch(error){
+    return res.status(500).send(error);
```
Comment (moshmage) [inline pages/api/nft/index.ts:102]: should log this to elastic

**Review by moshmage — CHANGES_REQUESTED** [review]

File: pages/api/nft/index.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    if(networkBounty.canceled || networkBounty.closed)
+      return res.status(404).json("Bounty has been closed or canceled");
+
+    const proposal = networkBounty.proposals.find(p=> p.id === +proposalscMergeId)
+    
+    if(!proposal)
+      return res.status(404).json("Proposal invalid");
+
+    if(proposal.refusedByBountyOwner || await network.isProposalDisputed(issueContractId, proposalscMergeId))
+      return res.status(404).json("proposal can be accepted");
```
Comment (vhcsilva) [inline pages/api/nft/index.ts:72]:
```suggestion
      return res.status(404).json("proposal cannot be accepted");
```

File: pages/api/nft/index.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+    const pullRequest = networkBounty.pullRequests.find(pr=> pr.id === proposal.prId)
+
+    if(pullRequest.canceled || !pullRequest.ready)
+      return res.status(404).json("PR can be accepted");
```
Comment (vhcsilva) [inline pages/api/nft/index.ts:77]:
```suggestion
      return res.status(404).json("PR cannot be accepted");
```

**Review by vhcsilva — APPROVED** [review]

**Review by moshmage — COMMENTED** [review]
lgtm, just replace the "can" for "cannot" like @vhcsilva noted

**Pull request merged**